src/socks-to-rtc/socks-to-rtc.spec.ts

```diff
@@ -100,6 +115,19 @@ describe('SOCKS server', function() {
     server.startInternal(mockTcpServer, mockPeerConnection).then(
         server.stop).then(onceServerStopped).then(done);
   });
+
+  it('stop works before the PeerConnection connects', (done) => {
+    (<any>mockTcpServer.onceListening).and.returnValue(Promise.resolve(mockEndpoint));
+    // PeerConnection never connects.
+    mockPeerConnection.onceConnected = noopPromise;
+    // Neither TCP connection nor datachannel close "naturally".
+    (<any>mockTcpServer.onceShutdown).and.returnValue(noopPromise);
+
+    var onceStartFailed :Promise<void> = new Promise<void>((F, R) => {
+      server.startInternal(mockTcpServer, mockPeerConnection).then(R, F);
+    });
+    Promise.all([onceStartFailed, server.stop(), onceServerStopped()]).then(done);
```

Comment:
So it's a bit weird to check both `server.stop` and `onceServerStopped`. Not wrong but I would leave out `onceServerStopped` and I think it would still be fine.